### PR TITLE
Refactor Cli to use standard Clap syntax

### PR DESCRIPTION
**Pull request by elliott-davis**
This DRYs up the cli code by moving bldr_url and friends into functions

Signed-off-by: Travis Elliott Davis <edavis@chef.io>

### CODE REVIEW

**Comment by thesentinels** [comment]
Thanks for the pull request! Here is what will happen next:
 1. Your PR will be reviewed by the maintainers
 2. If everything looks good, one of them will `approve` it, and your PR will be merged.

Thank you for contributing!

File: components/hab/src/cli.rs [hunk 1]
```diff
@@ -35,539 +35,618 @@ pub fn get() -> App<'static, 'static> {
...
-            (@setting ArgRequiredElseHelp)
-            (subcommand: sub_cli_setup().aliases(&["s", "se", "set", "setu"]))
-            (subcommand: sub_cli_completers().aliases(&["c", "co", "com", "comp"]))
+    App::new("hab")
+        .about("\"A Habitat is the natural environment for your services\" - Alan Turing")
+        .version(super::VERSION)
+        .author("\nAuthors: The Habitat Maintainers <humans@habitat.sh>\n")
+        .setting(AppSettings::VersionlessSubcommands)
+        .setting(AppSettings::ArgRequiredElseHelp)
+        .subcommand(SubCommand::with_name("clone")
```
Comment (raskchanky) [inline components/hab/src/cli.rs]: Is this supposed to be `clone` or should it be `cli`?

**Comment by raskchanky** [comment]
One minor nit but otherwise I love this change.  I've come around to the opinion that using macros to define cutesy DSLs is a bad idea and leads to code that's difficult to read and not flexible.  Pulling all those repeated bits out into functions feels so much better.

**Comment by elliott-davis** [comment]
@thesentinels approve

**Comment by thesentinels** [comment]
:metal: I am testing your branch against master before merging it. We do this to ensure that the master branch is never failing tests.

**Comment by thesentinels** [comment]
:neckbeard: Travis CI has [started testing this PR](https://travis-ci.org/habitat-sh/habitat/builds/338576613).

**Comment by thesentinels** [comment]
:sparkling_heart: Travis CI [reports this PR passed](https://travis-ci.org/habitat-sh/habitat/builds/338576613).

It always makes me feel nice when humans approve of one anothers work. I'm merging this PR now.

I just want you and the contributor to answer me one question:

gif-keyboard-3280869874741411265

**Pull request merged**